### ISSUE TITLE
Callback Functions not working on version 0.38.08

### ISSUE THREAD

**Issue by anjugajbhiye91** [niivue/niivue#819]
CallBack Function OnLocationChange not working.

My Application is pointing to index.d.ts while getting the below error

Error: onLocationChange doesnot exist in type 'Partial < NiiVueOptions>'

when i  delete the file index.d.ts  the application is pointing to index.js and callback function is working.

**Comment by anjugajbhiye91** [comment]
I am sharing the Repository

Link: https://bitbucket.org/gajbhiyeanju/niivue-token/

**Comment by hanayik** [comment]
@anjugajbhiye91 , this project repo has the same canvas style issue. You can fix it by removing the style attribute [here](https://bitbucket.org/gajbhiyeanju/niivue-token/src/98d4f9b1742edaf09c611a50c0405431168e6379/src/app/app.component.html#lines-2).

The `onLocationChange` callback is working without `index.t.ts` as you noted. I tested this by moving it like this: 

```
mv ./node_modules/@niivue/niivue/dist/index.d.ts ./node_modules/@niivue/niivue/dist/index.d.ts.bkup
```

Then updated the callback to log the argument passed in:

```
this.nv = new niivue.Niivue({
      isResizeCanvas: false,
      // crosshairWidth: 0,
      show3Dcrosshair: true,
      // backColor: [1, 1, 1, 1],
      backColor: [0.64, 0.76, 0.68, 1],
      // onLocationChange: this.handleLocationChange.bind(this),
      onLocationChange: (location: any) => {
        console.log(location)
      },
      // onImageLoaded: this.handleImageLoaded.bind(this),
    });
```

We'll need to investigate this more since there are a few lingering issues in the index.d.ts file pointed out by your angular build system (for example, another error is related to the fonts used to show the orientation labels in NiiVue).

**Comment by hanayik** [comment]
@jens-ox , our current build system (using `tsup`) creates `index.d.ts` and populates it with variables related to our JSON font metrics. This file is apparently invalid, and produces the warning: `Initializers are not allowed in ambient contexts.ts(1039)` in vscode (and the angular build tooling), which is referring to the fact that we (`tsup`) are initialising those font-related variables rather than defining their types in `index.d.ts`. Any ideas how we can configure this without a major rewrite to how we are dealing with the font metrics? JSON should be a viable loader type out-of-the-box for tsup, so I was surprised that it is generating an invalid file without warning at build time. 

example of offending lines in `index.d.ts`:

```javascript
var atlas = {
	type: "msdf",
	distanceRange: 2,
	size: 59.65625,
	width: 512,
	height: 256,
	yOrigin: "bottom"
};
```

**Comment by hanayik** [comment]
@jens-ox and @anjugajbhiye91 , this is partially fixed by PR #832 , which makes the font-related properties private, thus omitting them from the `index.d.ts` file. There may be more instances where this is needed to fix build errors or warning in other projects. The larger piece of work related to this will be #825.

**Comment by jens-ox** [comment]
@hanayik I manually checked the `index.d.ts` in #832, should be valid now. I don't know why tsup isn't able to properly generate declarations for JSON files, I'll investigate later (with low prio as #832 fixes this 😅)
